src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectLockFileWatcher.cs

```diff
@@ -159,18 +159,21 @@ public int FilesChanged(uint cChanges, string[] rgpszFile, uint[] rggrfChange)
             // the kind of change since we are interested in all changes.
             _projectServices.ThreadingService.Fork(async () =>
             {
-                using (var access = await _projectLockService.WriteLockAsync())
+                await _projectServices.Project.Services.ProjectAsynchronousTasks.LoadedProjectAsync(async () =>
```

Comment:
He's away on holidays, I've forwarded you a thread that takes about this. However, in this case, I don't think the ObjectDisposedException is caught further up.